tests/rptest/tests/consumer_group_test.py

```diff
@@ -98,33 +100,36 @@ def validate_group_state(self, group, expected_state, static_members):
             else:
                 assert p.instance_id is None
 
-    def setup_producer(self, p_cnt):
+    def create_topic(self, p_cnt):
         # create topic
         self.topic_spec = TopicSpec(partition_count=p_cnt,
                                     replication_factor=3)
+
         self.client().create_topic(specs=self.topic_spec)
+
+    def start_producer(self, msg_cnt=5000):
+
         # produce some messages to the topic
         self.producer = RpkProducer(self._ctx, self.redpanda,
-                                    self.topic_spec.name, 128, 5000, -1)
+                                    self.topic_spec.name, 128, msg_cnt, -1)
```

Comment:
Does this call wait for consumer-partition assignment to complete? If not, there is still a chance for a race (when one of consumers gets all the messages), albeit much smaller than before.

Reply:
we indeed do not wait for assignment, i will introduce explicit wait for all consumers to join the group